lib/LineFollower/ReadyState.cpp

```diff
@@ -1,142 +1,150 @@
+/**
+ * @brief  Calibration state
+ * @author Andreas Merkle <web@blue-andi.de>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "ReadyState.h"
+#include <Board.h>
+#include <StateMachine.h>
+#include "ReleaseTrackState.h"
+#include <Logging.h>
+
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+void ReadyState::entry()
+{
+    IDisplay&     display                 = Board::getInstance().getDisplay();
+    const int32_t SENSOR_VALUE_OUT_PERIOD = 1000; /* ms */
+
+    display.clear();
+    display.print("Rdy.");
+
+    if (true == m_isLapTimeAvailable)
+    {
+        display.gotoXY(0, 1);
+        display.print(m_lapTime);
+    }
+
+    /* The line sensor value shall be output on console cyclic. */
+    m_timer.start(SENSOR_VALUE_OUT_PERIOD);
+}
+
+void ReadyState::process(StateMachine& sm)
+{
+    IButton& buttonA = Board::getInstance().getButtonA();
+
+    /* Shall track be released? */
+    if (true == buttonA.isPressed())
+    {
+        buttonA.waitForRelease();
+        sm.setState(&ReleaseTrackState::getInstance());
+    }
+    /* Shall the line sensor values be printed out on console? */
+    else if (true == m_timer.isTimeout())
+    {
+        ILineSensors&   lineSensors  = Board::getInstance().getLineSensors();
+        uint8_t         index        = 0;
+        int16_t         position     = lineSensors.readLine();
+        const uint16_t* sensorValues = lineSensors.getSensorValues();
+        char            msg[80U]     = "";
+        char            tmp[10U]     = "";
+
+        /* Print line sensor value on console for debug purposes. */
+        for (index = 0; index < lineSensors.getNumLineSensors(); ++index)
+        {
+            if (0 < index)
+            {
+                strncat(msg, " / ", 4U);
```

Comment:
usually it shall ensure that the 3 characters can be safely added ;-)

Reply:
Would this be a better solution? 'strncat(msg, " / ", (sizeof(" / ") - 1))'

Reply:
No. The idea is to protect against a buffer overflow, means msg shall not be written out of bounce by strncat!

Solution:
```cpp
strncat(msg, " / ", sizeof(msg) - strlen(msg) - 1);
```